eclipse/src/saros/ui/command_handlers/SessionAddSelectedResourcesHandler.java

```diff
@@ -31,4 +48,21 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 
     return null;
   }
+
+  @CanExecute
+  public boolean canExecute(
+      @Named(IServiceConstants.ACTIVE_SELECTION) @Optional IResource resource) {
+    if (!(connectionHandler != null
+        && connectionHandler.isConnected()
+        && sessionManager != null
+        && sessionManager.getSession() != null
+        && sessionManager.getSession().hasWriteAccess())) {
+      return false;
+    }
+    final ISarosSession session = sessionManager.getSession();
```

Comment:
The session could be null at this point.

Reply:
Is there a better way besides the null check in lines 57/58? 
The Code is adapted from the ResourcePropertyTester where it is done with a simple null check as well.

Reply:
I think the problem is, that `getSession` might return a different object (possible null) on the second call, if the session is currently shutting down.

You need to first null-check the session manager, then call `getSession` and then null-check the session .